Source/Core/Core/IOS/ES/Views.cpp

```diff
@@ -98,17 +98,23 @@ IPCReply ESDevice::GetTicketViews(const IOCtlVRequest& request)
   return IPCReply(IPC_SUCCESS);
 }
 
-ReturnCode ESDevice::GetV0TicketFromView(const u8* ticket_view, u8* ticket) const
+ReturnCode ESDevice::GetTicketFromView(const u8* ticket_view, u8* ticket, u32* ticket_size) const
```

Comment:
This is mostly a theoretical concern, as I don't think there are titles that use both v0 and v1 tickets, but IMO we should check whether the ticket view version and the version of `installed_ticket` match.

If the view is v1 then we're only supposed to be looking for v1 tickets (and returning v1 tickets); v0 tickets should be ignored in that case and vice versa if the view is v0.

The easiest way to implement that would be to add a `std::optional<u8> desired_version` argument to FindSignedTicket that (if specified) constraints the version of tickets we can look up.

Reply:
I think it's worth having the ability to get a signed ticket without caring about its version; looking at ES, it looks like there's a generic "get ticket view" function that can generate a view from both v0 and v1 tickets, and that function is used for `ES_LaunchTitle`, the GetTicketViews IPC command and content checks among other things.